src/frontend/next/src/components/Post/Post.tsx

```diff
@@ -0,0 +1,199 @@
+import { useRef, useState } from 'react';
+
+import useSWR from 'swr';
+
+import { makeStyles, Theme } from '@material-ui/core/styles';
+import { Box, Grid, Typography, ListSubheader, createStyles } from '@material-ui/core';
+import { Post } from '../../interfaces';
+
+// For style, need to add to "index.tsx"
+// import '../styles/telescope-post-content.css';
+// See https://github.com/Seneca-CDOT/telescope/issues/1548 for discussion.
+
+type Props = {
+  postUrl: string;
+};
+
+const useStyles = makeStyles((theme: Theme) =>
+  createStyles({
+    root: {
+      padding: 0,
+      fontSize: '1.5rem',
+      marginBottom: '4em',
+    },
+    header: {
+      backgroundColor: theme.palette.primary.main,
+      color: theme.palette.text.secondary,
+      padding: '2em 3em 2em 3em',
+      lineHeight: '1.3',
+      zIndex: 1100,
+      top: '-1.1em',
+      [theme.breakpoints.down(1440)]: {
+        paddingTop: '1.6em',
+        paddingBottom: '1.5em',
+      },
+      [theme.breakpoints.down(1065)]: {
+        position: 'static',
+      },
+    },
+    expandHeader: {
+      whiteSpace: 'normal',
+      cursor: 'pointer',
+    },
+    collapseHeader: {
+      whiteSpace: 'nowrap',
+      cursor: 'pointer',
+    },
+    title: {
+      fontSize: '3.5em',
+      fontWeight: 'bold',
+      overflow: 'hidden',
+      textOverflow: 'ellipsis',
+      [theme.breakpoints.between('xs', 'sm')]: {
+        fontSize: '2.5em',
+      },
+    },
+    author: {
+      fontSize: '1.5em',
+      fontWeight: 'bold',
+      color: theme.palette.primary.contrastText,
+      [theme.breakpoints.between('xs', 'sm')]: {
+        fontSize: '1.2em',
+      },
+    },
+    published: {
+      fontSize: '1.2em',
+      textDecoration: 'none',
+      color: theme.palette.primary.contrastText,
+      [theme.breakpoints.between('xs', 'sm')]: {
+        fontSize: '1em',
+      },
+    },
+    content: {
+      overflow: 'auto',
+      padding: '2em',
+      color: theme.palette.text.primary,
+    },
+    link: {
+      textDecoration: 'none',
+      color: theme.palette.primary.contrastText,
+      '&:hover': {
+        textDecorationLine: 'underline',
+      },
+    },
+    time: {
+      '&:hover': {
+        textDecorationLine: 'underline',
+      },
+    },
+    spinner: {
+      padding: '20px',
+    },
+    error: {
+      lineHeight: '1.00',
+      fontSize: '1em',
+    },
+  })
+);
+
+const formatPublishedDate = (dateString: string) => {
+  const date = new Date(dateString);
+  const options = { month: 'long', day: 'numeric', year: 'numeric' };
+  const formatted = new Intl.DateTimeFormat('en-CA', options).format(date);
+  return `Last Updated ${formatted}`;
+};
+
+const Post = ({ postUrl }: Props) => {
```

Comment:
We can't have 2 things named Post.  You are importing a `Post` type, and now defining another `Post` as the component.  You could call this `PostComponent` if you want.